### DISCUSSION TITLE
T40 USB Cloner

### DISCUSSION THREAD

**Discussion by jvasanthan** [themactep/thingino-firmware#374] [General]
Hi, 
Have you tried to use USB cloner with T40? what are the steps to make it enter into USB boot mode.

**Comment by gtxaspec** [comment]
Same process as other Ingenic SoC's... 
![image](https://github.com/user-attachments/assets/2490377d-0b73-461b-a26c-d62e39da0242)

**Comment by gtxaspec** [comment]
moved to discussions for visibility...

**Comment by jvasanthan** [comment]
Thank you for your response. 

I watched one of your videos (https://www.youtube.com/watch?v=nqjejNizvC0) where you explained to let the bootloader boot and then short the pins on the SPI flash. If I have to make CPU enter into USB boot mode, should I do follow the same steps you explained in the video or should I short the pins even before CPU reading bootloader from flash. 
btw, I am planning to try this method in a PoE camera which doesn't have UART debug terminal.

**Comment by jvasanthan** [comment]
This video explained USB cloner. https://www.youtube.com/watch?v=nqjejNizvC0&t=2s
Cheers

**Comment by gtxaspec** [comment]
The typical process is to short the pins, apply power, and then release a few seconds later.  Since the SoC can not read the flash, it will proceed to boot into cloner mode.

**Comment by jvasanthan** [comment]
Will the technique work for both SPI and Quad SPI flash?

**Reply by gtxaspec** [reply]
it should, you need to short pins 5-6, whatever that is on the chip you are using, SCK and MOSI, on SOIC-8 is usually 5-6
